logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt

```diff
@@ -180,4 +183,12 @@ class DefaultPathAssertions : PathAssertions {
         other: String
     ): FeatureExtractorBuilder.ExecutionStep<T, Path> = container.f1<T, String, Path>(Path::resolve, other)
 
+    override fun <T : Path> hasDirectoryEntry(container: AssertionContainer<T>, entries: List<String>): Assertion =
+        assertionBuilder.invisibleGroup.withAssertions(
+            listOf(isDirectory(container)) +
```

Comment:
if the subject is not a directory, then all other assertions will fail as well. Maybe fail fast is better in this case? On the other hand, we get more context in the error report which might be nice as well.
Keep it this way